### ISSUE TITLE
Global Rate Limit does not work on istio 1.12.0

### ISSUE THREAD

**Issue by echinengk** [istio/istio#36284]
### Bug Description

I'm deploying Global Rate Limit using Istio version 1.12, but I'm having problems.
In the ingress gateway logs you can see that the Cluster (ratelimit) is not found
```
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337007Z	debug	envoy http	[C484] new stream
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337188Z	debug	envoy http	[C484][S1421517099832093952] request headers complete (end_stream=true):
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy ':authority', 'localhost:8080'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy ':path', '/productpage'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy ':method', 'GET'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'connection', 'keep-alive'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'pragma', 'no-cache'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'cache-control', 'no-cache'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-ch-ua', '" Not A;Brand";v="99", "Chromium";v="96", "Google Chrome";v="96"'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-ch-ua-mobile', '?0'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-ch-ua-platform', '"macOS"'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'upgrade-insecure-requests', '1'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'user-agent', 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.55 Safari/537.36'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'accept', 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-fetch-site', 'none'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-fetch-mode', 'navigate'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-fetch-user', '?1'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'sec-fetch-dest', 'document'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'accept-encoding', 'gzip, deflate, br'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'accept-language', 'en-US,en;q=0.9,pt;q=0.8'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337216Z	debug	envoy http	[C484][S1421517099832093952] request end stream
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337412Z	debug	envoy filter	rate limit fail, status=14 msg=Cluster not available
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337430Z	debug	envoy filter	rate limit status, status=1
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337449Z	debug	envoy http	[C484][S1421517099832093952] Sending local reply with details rate_limiter_error
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337494Z	debug	envoy http	[C484][S1421517099832093952] encoding headers via codec (end_stream=true):
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy ':status', '500'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'date', 'Mon, 29 Nov 2021 16:53:23 GMT'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 'server', 'istio-envoy'
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337738Z	debug	envoy wasm	wasm log stats_outbound stats_outbound: [extensions/stats/plugin.cc:640]::report() metricKey cache hit , stat=12
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337785Z	debug	envoy wasm	wasm log stats_outbound stats_outbound: [extensions/stats/plugin.cc:640]::report() metricKey cache hit , stat=6
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337793Z	debug	envoy wasm	wasm log stats_outbound stats_outbound: [extensions/stats/plugin.cc:640]::report() metricKey cache hit , stat=10
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy 2021-11-29T16:53:23.337818Z	debug	envoy wasm	wasm log stats_outbound stats_outbound: [extensions/stats/plugin.cc:640]::report() metricKey cache hit , stat=14
istio-ingressgateway-55d9fb9f-mn7jk istio-proxy [2021-11-29T16:53:23.337Z] "GET /productpage HTTP/1.1" 500 - rate_limiter_error - "-" 0 0 0 - "172.17.0.5" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.55 Safari/537.36" "cff85b2b-5edf-92e4-8b9e-ad77c1212226" "localhost:8080" "-" outbound|9080||productpage.default.svc.cluster.local - 127.0.0.1:8080 127.0.0.1:50648 - -
```

I used this Doc to validate the functionality: https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#global-rate-limit

I used the same DOC with version 1.11.4 and I didn't have any problems!

Is anyone else having the same problem?

### Version

```prose
λ istioctl version
client version: 1.12.0
control plane version: 1.12.0
data plane version: 1.12.0 (10 proxies)

λ kubectl version --short
Client Version: v1.21.5
Server Version: v1.22.3
```

### Additional Information

```
λ istioctl bug-report

Target cluster context: minikube

Running with the following config:

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-system,kube-public,kube-node-lease,local-path-storage }
end-time: 2021-11-29 13:59:16.508634 -0300 -03



Cluster endpoint: https://127.0.0.1:54376
CLI version: version.BuildInfo{Version:"1.12.0", GitRevision:"016bc46f4a5e0ef3fa135b3c5380ab7765467c1a", GolangVersion:"go1.17.3", BuildStatus:"Clean", GitTag:"1.12.0"}

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.12.0", GitRevision:"016bc46f4a5e0ef3fa135b3c5380ab7765467c1a", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.12.0"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.12.0}


Fetching proxy logs for the following containers:

default/details-v1/details-v1-79f774bdb9-ns4w9/details
default/details-v1/details-v1-79f774bdb9-ns4w9/istio-proxy
default/productpage-v1/productpage-v1-6b746f74dc-sb5nk/istio-proxy
default/productpage-v1/productpage-v1-6b746f74dc-sb5nk/productpage
default/ratelimit/ratelimit-5d7db5b95-h7wmh/istio-proxy
default/ratelimit/ratelimit-5d7db5b95-h7wmh/ratelimit
default/ratelimit/ratelimit-5d7db5b95-h7wmh/statsd-exporter
default/ratings-v1/ratings-v1-b6994bb9-bjcjv/istio-proxy
default/ratings-v1/ratings-v1-b6994bb9-bjcjv/ratings
default/redis/redis-7d757c948f-lx5dz/istio-proxy
default/redis/redis-7d757c948f-lx5dz/redis
default/reviews-v1/reviews-v1-545db77b95-8dpds/istio-proxy
default/reviews-v1/reviews-v1-545db77b95-8dpds/reviews
default/reviews-v2/reviews-v2-7bf8c9648f-rc5td/istio-proxy
default/reviews-v2/reviews-v2-7bf8c9648f-rc5td/reviews
default/reviews-v3/reviews-v3-84779c7bbc-thkzr/istio-proxy
default/reviews-v3/reviews-v3-84779c7bbc-thkzr/reviews
istio-system/istio-egressgateway/istio-egressgateway-7f4864f59c-zb2xb/istio-proxy
istio-system/istio-ingressgateway/istio-ingressgateway-55d9fb9f-mn7jk/istio-proxy
istio-system/istiod/istiod-555d47cb65-wpqcd/discovery

Fetching Istio control plane information from cluster.

Running istio analyze on all namespaces and report as below:
Analysis Report:

Creating an archive at /Users/elvischinen/bug-report.tar.gz.
Cleaning up temporary files in /var/folders/78/5hlm1f8j6v7_r3jb88fkqqkr0000gn/T/bug-report.
Done.
```

**Comment by bianpengyuan** [comment]
This is broken by https://github.com/istio/istio/pull/35718. In the rate limit doc we don't specify EnvoyFilter patch context. I think the dedup fix at https://github.com/istio/istio/pull/35718 should also not skip if it is to patch a gateway.

**Comment by kraft87** [comment]
What is the fix for this? Is it to set EnvoyFilter match context to ANY in the second patch?

**Comment by bianpengyuan** [comment]
You need to set it to `GATEWAY` in the example case.

**Comment by bianpengyuan** [comment]
With the fix, you don't need to set anything. The fix will be shipped with 1.12.1
